components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -2880,4 +2883,43 @@ private String getIDPNameByMetadataProperty(Connection dbConnection, String prop
 
         return idPName;
     }
+
+    /**
+     * Validate whether the outbound provisioning roles does exist.
+     *
+     * @param identityProvider IdentityProvider.
+     * @param tenantDomain     Tenant Domain.
+     * @throws IdentityProviderManagementException If an error occurred while checking for role existence.
+     */
+    private void validateOutboundProvisioningRoles(IdentityProvider identityProvider, String tenantDomain)
+            throws IdentityProviderManagementException {
+
+        String provisioningRole = identityProvider.getProvisioningRole();
+        if (StringUtils.isBlank(provisioningRole)) {
+            return;
+        }
+        String[] outboundProvisioningRoles = StringUtils.split(provisioningRole, ",");
+
+        try {
+            String notExistingRoles = null;
+            RoleManagementService roleManagementService =
+                    IdpMgtServiceComponentHolder.getInstance().getRoleManagementService();
+            for (String roleName : outboundProvisioningRoles) {
+                if (StringUtils.isNotBlank(notExistingRoles)) {
+                    notExistingRoles = notExistingRoles + ",";
+                }
+                if (!roleManagementService.isExistingRoleName(roleName, tenantDomain)) {
+                    notExistingRoles = roleName;
```

Comment:
I assume you want to show the list of non existing roles in the error response. In that case, this logic seems not to work when 2+ non-existing roles are given. In the 2nd iteration this will replace "1st_role," with "2nd_role" instead of appending

Reply:
Fixed: 3b8457ce1a80e96bdf0458aea59907543b13f5ba